src/pseudopeople/noise_functions.py

```diff
@@ -360,23 +360,28 @@ def write_wrong_digits(
     rng = np.random.default_rng(get_hash(f"{randomness_stream.seed}_write_wrong_digits"))
     column = column.astype(str)
     max_str_length = column.str.len().max()
-    same_len_col = column.str.pad(max_str_length, side="right")
-    is_number = pd.concat(
-        [same_len_col.str[i].str.isdigit() for i in range(max_str_length)], axis=1
+
+    possible_replacements = np.array(list("0123456789"))
+
+    # https://stackoverflow.com/a/9493192/
+    # Changing this to a numpy (not Python) string type means that it will have a fixed
+    # number of characters, equal to the longest string in the array.
+    # view("U1") then reinterprets this memory as an array of individual (Unicode) characters.
+    same_len_col_exploded = (
+        column.values.astype(str).view("U1").reshape((len(column), max_str_length))
+    )
+    # NOTE: np.isin does not work with sets, see https://numpy.org/doc/stable/reference/generated/numpy.isin.html
+    is_number = (
+        pd.DataFrame(same_len_col_exploded).isin(set(possible_replacements)).to_numpy()
```

Comment:
Does `same_len_col_exploded.isin(possible_replacements)` not work? Aren't they both `np.arrays`?

Reply:
You're right, they are. For whatever reason, NumPy's .isin is orders of magnitude slower here. I'll update this comment accordingly.